lib/blockchain/chain.js

```diff
@@ -935,8 +935,10 @@ class Chain extends AsyncEmitter {
       const ns = await view.getNameState(this.db, nameHash);
 
       if (ns.isNull()) {
-        if (!covenant.isClaim() && !covenant.isOpen())
+        if (!covenant.isClaim() && !covenant.isOpen()) {
+          this.emit('CRITICAL ERROR', 'Unexpected null NameState.');
           throw new Error('Database inconsistency.');
```

Comment:
Maybe for the critical errors, using different Error class would be more useful in the future if we want to identify those.

e.g.

```js
class CriticalError extends Error {
  constructor(module, msg) {
    super();

    assert(typeof msg === 'string');

    this.type = 'CriticalError';
    this.module = module;
    this.message = msg;

    if (Error.captureStackTrace)
      Error.captureStackTrace(this, CriticalError);
  }
}

throw new CriticalError('chain', 'Database inconsistency');

```

Reply:
Where do you think `CriticalError` class should live? We don't have a global module for these, but this type of error could come from anywhere